enhancing/enhancer.js

```diff
@@ -2,21 +2,41 @@ module.exports = {
   succeed,
   fail,
   repair,
-  get,
+  get
 };
 
 function succeed(item) {
-  return { ...item };
+  return {
+    ...item,
+    enhancement: item.enhancement < 20 ? item.enhancement + 1 : 20
+  };
 }
 
 function fail(item) {
-  return { ...item };
+  let itemCopy = { ...item };
+  if (item.enhancement < 15) {
+    itemCopy.durability -= 5;
+  }
+  if (item.enhancement > 15) {
+    itemCopy.durability -= 10;
+  }
+  if (item.enhancement > 16) {
+    itemCopy.enhancement -= 1;
+  }
```

Comment:
I guess I don't understand. My line of reasoning is that there still needs to be at least an else there to make sure the code branches correctly?

Reply:
This series of checks simulate your three `if`s, just with the conditions themselves being considered as well so that we don't run through all three checks if we already know that we don't need to. It might be different if the thing being checked changed between the conditions, but `item.enhancement` (and `item` itself) isn't ever altered here.

It doesn't make a big difference (it's probably not even noticeable at this scale), but it does add up if you have a full application built out. We want our tests to run fast, and that includes the things that we're testing as well!

Reply:
Thanks! I was really confused.